ext/shmop/shmop.c

```diff
@@ -178,35 +178,35 @@ PHP_FUNCTION(shmop_open)
 			*/
 			break;
 		default:
-			php_error_docref(NULL, E_WARNING, "Invalid access mode");
+			zend_argument_value_error(2, "must be a valid access mode");
 			goto err;
 	}
 
 	if (shmop->shmflg & IPC_CREAT && shmop->size < 1) {
-		php_error_docref(NULL, E_WARNING, "Shared memory segment size must be greater than zero");
+		zend_argument_value_error(4, "must be greater than 0 for the \"c\" and \"n\" access modes");
 		goto err;
 	}
 
 	shmop->shmid = shmget(shmop->key, shmop->size, shmop->shmflg);
 	if (shmop->shmid == -1) {
-		php_error_docref(NULL, E_WARNING, "Unable to attach or create shared memory segment '%s'", strerror(errno));
+		php_error_docref(NULL, E_WARNING, "Unable to attach or create shared memory segment \"%s\"", strerror(errno));
 		goto err;
 	}
 
 	if (shmctl(shmop->shmid, IPC_STAT, &shm)) {
 		/* please do not add coverage here: the segment would be leaked and impossible to delete via php */
-		php_error_docref(NULL, E_WARNING, "Unable to get shared memory segment information '%s'", strerror(errno));
+		php_error_docref(NULL, E_WARNING, "Unable to get shared memory segment information \"%s\"", strerror(errno));
 		goto err;
 	}
 
 	if (shm.shm_segsz > ZEND_LONG_MAX) {
-		php_error_docref(NULL, E_WARNING, "shared memory segment too large to attach");
+		php_error_docref(NULL, E_WARNING, "Shared memory segment too large to attach");
```

Comment:
I couldn't decide if this should be promoted or not

Reply:
Should be fine to promote.